scripts/travis/external_build.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+# external_build.sh - Enqueue the build request for the specified platform
+#
+# Syntax:   external_build.sh
+#
+# Usage:    Should only be used by Travis
+#
+# Examples: scripts/travis/external_build.sh
+#
+# Use the following travis varaibles:
+# BUILD_TYPE
+# TARGET_PLATFORM
+# TRAVIS_BRANCH
+# TRAVIS_COMMIT
+# TRAVIS_BUILD_NUMBER
+# BUILD_REQUESTS_BUCKET
+# BUILD_PULL_REQUESTS_BUCKET
+# AWS_ACCESS_KEY_ID (optional)
+# AWS_SECRET_ACCESS_KEY (optional)
+#
+
+# exit on error
+set -e
+
+BUILD_TARGET=$1
+
+if [ "${BUILD_TYPE}" != "external_build" ]; then
+    echo "error: wrong build type specified '${BUILD_TYPE}'"
+    exit 1
+fi
+
+if [ "${TARGET_PLATFORM}" != "linux-arm64" ] && [ "${TARGET_PLATFORM}" != "linux-arm" ]; then
+    echo "error: unsupported platform '${TARGET_PLATFORM}'"
+    exit 1
+fi
+
+if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then
+    if [ "${BUILD_REQUESTS_BUCKET}" = "" ]; then
+        echo "error: BUILD_REQUESTS_BUCKET was not specified."
+        exit 1
+    fi
+else
+    if [ "${BUILD_PULL_REQUESTS_BUCKET}" = "" ]; then
+        echo "error: BUILD_PULL_REQUESTS_BUCKET was not specified."
+        exit 1
+    fi
+fi
+
+if [ "${BUILD_TARGET}" = "" ]; then
+    echo "error: missing build target argument"
+    exit 1
+fi
+
+sudo apt-get install awscli
+
+# create build request
+echo "{ \"TRAVIS_BRANCH\" : \"${TRAVIS_BRANCH}\", \"TRAVIS_COMMIT\" : \"${TRAVIS_COMMIT}\", \"TRAVIS_PULL_REQUEST\" : \"${TRAVIS_PULL_REQUEST}\", \"AWS_ACCESS_KEY_ID\" : \"${AWS_ACCESS_KEY_ID}\", \"AWS_SECRET_ACCESS_KEY\" : \"${AWS_SECRET_ACCESS_KEY}\", \"EXEC\":\"${BUILD_TARGET}\" }" > ${TRAVIS_BUILD_NUMBER}.json
```

Comment:
makes sense to output params to travis log as well

Reply:
noted; I'll add this in a subsequent PR.